app/scripts/modules/core/src/managed/environmentBaseElements/EnvironmentsRender.tsx

```diff
@@ -0,0 +1,83 @@
+import classnames from 'classnames';
+import { reverse } from 'lodash';
+import React from 'react';
+import { atom, useRecoilState } from 'recoil';
+
+import { useElementDimensions } from 'core/presentation/hooks/useDimensions.hook';
+import { logger } from 'core/utils';
+
+const STORAGE_KEY = 'MD_environmentsDirection';
+
+const DIRECTIONS = ['list', 'sideBySide'] as const;
+type Direction = typeof DIRECTIONS[number];
+
+const isDirection = (value: string | null): value is Direction => {
+  return Boolean(value && DIRECTIONS.includes(value as Direction));
+};
+
+const storedDirection = localStorage.getItem(STORAGE_KEY);
+
+const environmentsDirectionState = atom<Direction>({
+  key: 'environmentsDisplay',
+  default: isDirection(storedDirection) ? storedDirection : 'list',
+});
+
+// The goal of this hook is to store the value in an atom to be shared across the app but also update the local storage
+const useEnvironmentDirection = () => {
+  const [direction, setDirection] = useRecoilState(environmentsDirectionState);
+  React.useLayoutEffect(() => {
```

Comment:
I can't think of a reason  why useLayoutEffect would be appropriate here (as opposed to regular old useEffect).  Can you explain?

Reply:
You can't think of a reason because there is no reason :)